src/libertem/web/connect.py

```diff
@@ -2,22 +2,32 @@
 from functools import partial
 
 import tornado.web
-
 from libertem.executor.base import AsyncAdapter, sync_to_async
 from libertem.executor.dask import DaskJobExecutor, cluster_spec
+from libertem.utils.devices import detect
+
+from .base import ResultHandlerMixin, SessionsHandler, log_message
 from .messages import Message
-from .base import log_message, ResultHandlerMixin
 from .state import SharedState
-from libertem.utils.devices import detect
 
 log = logging.getLogger(__name__)
 
+def connect_to_local(state, connection, pool):
```

Comment:
I separate the connection code to call it from different places as required for the two modes we now have.